### PR TITLE
document upload and download now works

### PR DESCRIPTION
**Pull request by HLaland**
- Added functionality to add document to item
- Added functionality to add document to item template
- GetByItemId returns document for item and its item template
- DeleteDocument deletes a document from an item or template

### CODE REVIEW

File: Controllers/DocumentController.cs [hunk 1]
```diff
@@ -12,18 +12,37 @@ public class DocumentController : ControllerBase
...
+            {
+                return BadRequest($"Something went wrong: {e.Message}");
+            }
+        }
+
+
         [HttpGet("ByItemId/{id}")]
-        [SwaggerOperation(Summary = "Get documentation by item ID", Description = "Retrieves all files related to an item")]
-        [SwaggerResponse(200, "Success", typeof(IEnumerable<Document>))]
+        [SwaggerOperation(Summary = "Get documentation by item ID", Description = "Retrives all files related to an item")]
```
Comment (eikib) [inline Controllers/DocumentController.cs:42]: og du skylde på min engelsk? Retrieves

File: Models/Document/DocumentUploadDto.cs
Comment (eikib) [inline Models/Document/DocumentUploadDto.cs:1]: ska ikkje upload ha f eks item id eller template id?

**Review by eikib — COMMENTED** [review]
fiks null warnings

File: Models/Document/DocumentUploadDto.cs [hunk 2]
```diff
@@ -2,7 +2,6 @@ namespace Inventory.Models.DocumentDtos;
 
 public class DocumentUploadDto
 {
-    public string? ItemId { get; set; }
```
[the remark this answers is not in the document]
Comment (HLaland) [reply Models/Document/DocumentUploadDto.cs:5]: nei, blir sendt i URL

**Review by eikib — APPROVED** [review]

**Pull request merged**